broker-core/src/main/java/io/zeebe/broker/subscription/message/processor/OpenMessageSubscriptionProcessor.java

```diff
@@ -17,25 +17,81 @@
  */
 package io.zeebe.broker.subscription.message.processor;
 
+import static io.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import io.zeebe.broker.logstreams.processor.SideEffectProducer;
 import io.zeebe.broker.logstreams.processor.TypedRecord;
 import io.zeebe.broker.logstreams.processor.TypedRecordProcessor;
 import io.zeebe.broker.logstreams.processor.TypedResponseWriter;
 import io.zeebe.broker.logstreams.processor.TypedStreamWriter;
+import io.zeebe.broker.subscription.command.SubscriptionCommandSender;
 import io.zeebe.broker.subscription.message.data.MessageSubscriptionRecord;
+import io.zeebe.broker.subscription.message.state.MessageDataStore;
+import io.zeebe.broker.subscription.message.state.MessageDataStore.Message;
+import io.zeebe.broker.subscription.message.state.MessageSubscriptionDataStore;
+import io.zeebe.logstreams.processor.EventLifecycleContext;
 import io.zeebe.protocol.intent.MessageSubscriptionIntent;
+import java.util.function.Consumer;
+import org.agrona.DirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
 
 public class OpenMessageSubscriptionProcessor
     implements TypedRecordProcessor<MessageSubscriptionRecord> {
 
+  private final MessageDataStore messageStore;
+  private final MessageSubscriptionDataStore subscriptionStore;
+
+  private final DirectBuffer messagePayload = new UnsafeBuffer(0, 0);
```

Comment:
Just out of curiosity, what's the difference between doing `new UnsafeBuffer()` (which just returns the same as `new UnsafeBuffer(ArrayUtil.EMPTY_BYTE_ARRAY)` where `EMPTY_BYTE_ARRAY = new byte[0]`) and `new UnsafeBuffer(0, 0)`? I get that the address it points to is different (i.e. the latter essentially being nullptr), but other than that, is there any reason to prefer one over the other?

Reply:
I guess that there is no difference. Usually, we use `new UnsafeBuffer(0,0)` when we only want to wrap another byte array (don't allocate something).